src/legacy/ui/public/vis/editors/default/components/default_editor_agg_select.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { EuiComboBox, EuiFormRow, EuiLink } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { get } from 'lodash';
+import React from 'react';
+import { AggType } from 'ui/agg_types';
+import { AggConfig } from 'ui/vis/agg_config';
+import { ComboBoxGroupedOption } from '../default_editor_utils';
+
+interface DefaultEditorAggSelectProps {
+  agg: AggConfig;
+  value: AggType;
+  setValue: (aggType: AggType) => void;
+  aggHelpLink: string;
+  aggTypeOptions: AggType[];
+  isSubAggregation: boolean;
+  isSelectInvalid: boolean;
+}
+
+function DefaultEditorAggSelect({
+  // since it happens that during 'agg_params' test run, this component is invoked with undefined props,
+  // we added default value for agg, value and aggTypeOptions. It can be removed after 'agg_params' is converted to React
+  agg = {},
+  value = { title: '' },
+  setValue,
+  aggTypeOptions = [],
+  aggHelpLink,
+  isSelectInvalid,
+  isSubAggregation,
+}: DefaultEditorAggSelectProps) {
+  const isAggTypeDefined = value && Boolean(value.title);
+  const selectedOptions: ComboBoxGroupedOption[] = isAggTypeDefined
+    ? [{ label: value.title, value }]
+    : [];
+
+  const label = isSubAggregation ? (
+    <FormattedMessage
+      id="common.ui.vis.defaultEditor.aggSelect.subAggregationLabel"
+      defaultMessage="Sub aggregation"
+    />
+  ) : (
+    <FormattedMessage
+      id="common.ui.vis.defaultEditor.aggSelect.aggregationLabel"
+      defaultMessage="Aggregation"
+    />
+  );
+  const helpLink = isAggTypeDefined && aggHelpLink && (
+    <EuiLink
+      href={aggHelpLink}
+      target="_blank"
+      rel="noopener"
+      className="visEditorAggSelect__helpLink pull-right"
+      type="button"
```

Comment:
This is the wrong type as it has an `href` and `EuiLink` should handle adding the appropriate `type` attribute.
```suggestion
```